src/goLiveErrors.ts

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+import vscode = require('vscode');
+import { byteOffsetAt, getBinPath } from './util';
+import cp = require('child_process');
+import path = require('path');
+import { promptForMissingTool } from './goInstallTools';
+import { diagnosticCollection } from './goMain';
+
+// Interface for settings configuration for adding and removing tags
+interface GoLiveErrorsConfig {
+	delay: number;
+	enabled: boolean;
+}
+
+let runner;
+
+// parseLiveFile runs the gotype command in live mode to check for any syntactic or
+// semantic errors and reports them immediately
+export function parseLiveFile(e: vscode.TextDocumentChangeEvent) {
+	if (e.document.isUntitled) {
+		return;
+	}
+	if (e.document.languageId !== 'go') {
+		return;
+	}
+
+	let config = <GoLiveErrorsConfig>vscode.workspace.getConfiguration('go')['liveErrors'];
```

Comment:
I use `config.delay` a few lines below, otherwise I would have reused that function, yeah.